From cfe3c76a4dfdd1dd8c3c574b913f6a76c917a1b6 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Fri, 4 Nov 2016 15:47:22 +0800
Subject: [PATCH 1336/5242] MLK-13422: ASoC: wm8960: fix the pitch shift issue
 after suspend/resume

commit  3a3b178d8db49622f45023ca81820ec55326ace6 from
https://source.codeaurora.org/external/imx/linux-imx.git

Before suspend, the sysclock select pll out as source, after resume
sysclock select the mlck as the source, so the sample rate is changed
the sound's pitch is shifted.
The issue is caused by the commit c1845da3d08de6cf2642fec74f7a46d05de6314d
Which removed the wm8960_configure_clocking() when bias level changes
from STANDBY to PREPARE, this patch is to add it back.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/wm8960.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/codecs/wm8960.c b/sound/soc/codecs/wm8960.c
index 7b6f709..db7a327 100644
--- a/sound/soc/codecs/wm8960.c
+++ b/sound/soc/codecs/wm8960.c
@@ -915,6 +915,10 @@ static int wm8960_set_bias_level_out3(struct snd_soc_component *component,
 				}
 			}
 
+			ret = wm8960_configure_clocking(codec);
+			if (ret)
+				return ret;
+
 			/* Set VMID to 2x50k */
 			snd_soc_component_update_bits(component, WM8960_POWER1, 0x180, 0x80);
 			break;
-- 
1.7.9.5

